packages/glob/src/glob.ts

```diff
@@ -0,0 +1,189 @@
+import * as core from '@actions/core'
+import * as fs from 'fs'
+import * as path from 'path'
+import * as patternHelper from './internal-pattern-helper'
+import {IGlobOptions} from './internal-glob-options'
+import {MatchResult} from './internal-match-result'
+import {Pattern} from './internal-pattern'
+import {SearchState} from './internal-search-state'
+
+export {IGlobOptions}
+
+/**
+ * Returns files and directories matching the specified glob pattern.
+ *
+ * Order of the results is not guaranteed.
+ */
+export async function glob(
+  pattern: string,
+  options?: IGlobOptions
+): Promise<string[]> {
+  // Set defaults options
+  options = getOptions(options)
+
+  // Parse patterns
+  const patterns: Pattern[] = patternHelper.parse([pattern], options)
+
+  // Get search paths
+  const searchPaths: string[] = patternHelper.getSearchPaths(patterns)
+
+  // Search
+  const result: string[] = []
+  for (const searchPath of searchPaths) {
+    // Skip if not exists
+    try {
+      await fs.promises.lstat(searchPath)
```

Comment:
Is there an advantage to this over using `fs.existsSync(searchPath)`?

If so, is it still valid to check the symbolic link (`lstat`) of the path instead of the file itself (`stat`)?

Reply:
i'll add a comment to the code to explain this. not sure if i even need this anymore actually

Reply:
i added a comment to the code. But we still need this `lstat` call here because `stat` is used later to determine whether the symlink is broken

Reply:
i think i can rearrange things a bit to make this more clean

Reply:
rearranged to eliminate some unnecessary stat calls (for files that arent matched)

also i eliminated some unnecessary nesting of loops by pre-populating the search stack with all search paths